packages/contracts-bedrock/src/L2/SuperchainERC20.sol

```diff
@@ -0,0 +1,126 @@
+    /// @param amount  Amount of tokens sent.
+    /// @param chainId Chain ID of the destination chain.
+    /// @param data    Data to be sent with the message.
+    event SentERC20(address indexed from, address indexed to, uint256 amount, uint256 chainId, bytes data);
+
+    /// @notice Emitted whenever tokens are successfully relayed on this chain.
+    /// @param to     Address of the recipient.
+    /// @param amount Amount of tokens relayed.
+    /// @param data   Data sent with the message.
+    event RelayedERC20(address indexed to, uint256 amount, bytes data);
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @param _messenger   Address of the L2ToL2CrossDomainMessenger.
+    /// @param _name        ERC20 name.
+    /// @param _symbol      ERC20 symbol.
+    /// @param _decimals    ERC20 decimals.
+    constructor(
+        address _messenger,
+        string memory _name,
+        string memory _symbol,
+        uint8 _decimals
+    )
+        ERC20(_name, _symbol)
+    {
+        MESSENGER = _messenger;
+        DECIMALS = _decimals;
+    }
+
+    /// @notice Sends tokens to some target address on another chain.
+    /// @param _to      Address to send tokens to.
+    /// @param _amount  Amount of tokens to send.
+    /// @param _chainId Chain ID of the destination chain.
+    /// @param _data    Data to be sent with the message.
+    function sendERC20(address _to, uint256 _amount, uint256 _chainId, bytes memory _data) external override {
+        _burn(msg.sender, _amount);
+
+        bytes memory _message = abi.encodeCall(this.relayERC20, (_to, _amount, _data));
+        IL2ToL2CrossDomainMessenger(MESSENGER).sendMessage(_chainId, address(this), _message);
+
+        emit SentERC20(msg.sender, _to, _amount, _chainId, _data);
+    }
+
+    /// @notice Relays tokens received from another chain.
+    /// @param _to     Address to relay tokens to.
+    /// @param _amount Amount of tokens to relay.
+    /// @param _data   Data sent with the message.
+    function relayERC20(address _to, uint256 _amount, bytes memory _data) external override {
+        if (msg.sender != MESSENGER) revert RelayMessageCallerNotL2ToL2CrossDomainMessenger();
+
+        if (IL2ToL2CrossDomainMessenger(MESSENGER).crossDomainMessageSender() != address(this)) {
+            revert MessageSenderNotThisSuperchainERC20();
+        }
+
+        _mint(_to, _amount);
+
+        if (_data.length > 0) {
+            bool success = SafeCall.call(_to, 0, _data);
+            if (!success) revert ExternalCallFailed();
+        }
+
+        emit RelayedERC20(_to, _amount, _data);
+    }
+
+    /// @notice ERC165 interface check function.
+    /// @param _interfaceId Interface ID to check.
+    /// @return Whether or not the interface is supported by this contract.
+    function supportsInterface(bytes4 _interfaceId) external pure virtual returns (bool) {
```

Comment:
let's remove this for now, ideally if things are done well, we won't need it and if we do it can be added in an upgrade